client/app/components/packages/landuse-form/proposed-action-editor.js

```diff
@@ -1,13 +1,36 @@
 import Component from '@glimmer/component';
+import { action } from '@ember/object';
 import { optionset } from '../../../helpers/optionset';
 
 export default class PackagesLanduseFormProposedActionEditorComponent extends Component {
   actionCode;
 
+  requiredActionCodes = ['CM', 'LD', 'RA', 'RC', 'RS', 'SA', 'SC', 'SD', 'ZA', 'ZC', 'ZS'];
+
+  get projectHasRequiredActions() {
+    const actionCode = this.args.landuseActionForm.data.dcpActioncode;
+    return this.requiredActionCodes.includes(actionCode);
+  }
+
+  get projectHasRequiredActionsAndFollowUpYes() {
+    const hasPreviousAction = this.args.landuseActionForm.data.dcpFollowuptopreviousaction || this.args.landuseActionForm.data.dcpPreviouslyapprovedactioncode;
+    return this.projectHasRequiredActions && hasPreviousAction;
+  }
+
   get selectedActionPlaceholder() {
     const previouslyApprovedActionCode = this.args.landuseActionForm.data.dcpPreviouslyapprovedactioncode;
     if (previouslyApprovedActionCode) {
       return optionset(['landuseAction', 'dcpPreviouslyapprovedactioncode', 'label', previouslyApprovedActionCode]);
     } return '-- select an action --';
   }
+
+  @action
+  setPreviouslyApprovedActionCodeFields(actionCode) {
+    if (actionCode) {
+      this.args.landuseActionForm.data.dcpPreviouslyapprovedactioncode = actionCode;
+      this.args.landuseActionForm.data.dcpFollowuptopreviousaction = true;
```

Comment:
So this action was created to make sure that  `dcpFollowuptopreviousaction` is also set when user makes a dropdown selection? What is the purpose of `dcpFollowuptopreviousaction`?

Reply:
Yes it was -- `dcpFollowuptopreviousaction` is required for display logic -- not sure it's totallyyy necessary (like we could just check that `dcpPreviouslyapprovedactioncode` is not a blank string or undefined but `dcpFollowuptopreviousaction` was included in the wireframe so I went ahead and incldued it here